pkg/certmanager/certmanager.go

```diff
@@ -211,3 +274,187 @@ func (m *manager) waitForCertificateRequest(ctx context.Context, log logr.Logger
 		}
 	}
 }
+
+const (
+	issuerNameKey  = "issuer-name"
+	issuerKindKey  = "issuer-kind"
+	issuerGroupKey = "issuer-group"
+)
+
+func (m *manager) handleRuntimeConfigIssuerChange(logger logr.Logger, event watch.Event) error {
+	m.activeIssuerRefMutex.Lock()
+	defer m.activeIssuerRefMutex.Unlock()
+
+	cm, ok := event.Object.(*corev1.ConfigMap)
+	if !ok {
+		return fmt.Errorf("got unexpected type for runtime configuration source; this is likely a programming error")
+	}
+
+	issuerRef := &cmmeta.ObjectReference{}
+
+	var dataErrs []error
+	var exists bool
+
+	issuerRef.Name, exists = cm.Data[issuerNameKey]
+	if !exists || len(issuerRef.Name) == 0 {
+		dataErrs = append(dataErrs, fmt.Errorf("missing key/value in ConfigMap data: %s", issuerNameKey))
+	}
+
+	issuerRef.Kind, exists = cm.Data[issuerKindKey]
+	if !exists || len(issuerRef.Kind) == 0 {
+		dataErrs = append(dataErrs, fmt.Errorf("missing key/value in ConfigMap data: %s", issuerKindKey))
+	}
+
+	issuerRef.Group, exists = cm.Data[issuerGroupKey]
+	if !exists || len(issuerRef.Group) == 0 {
+		dataErrs = append(dataErrs, fmt.Errorf("missing key/value in ConfigMap data; %s", issuerGroupKey))
+	}
+
+	if len(dataErrs) > 0 {
+		return errors.Join(dataErrs...)
+	}
+
+	// we now have a full issuerRef
+	// TODO: check if the issuer exists by querying for the CRD?
```

Comment:
note: This TODO is copied from csi-driver-spiffe - this feature merged in csi-driver-spiffe without this check.